### PR TITLE
Be more precise about where to read the URL rewriter config from

### PR DESCRIPTION
**Pull request by shs96c**
There is an [open issue](https://github.com/bazelbuild/bazel/issues/22104) where bazel will occasionally fail at start up time because it is unable to parse the downloader config. While I've yet to create a reliable reproduction for this issue, the one time I've triggered this error with a build with additional logging, it seemed to be because the current working directory wasn't the root of the workspace. While I'm not sure this will actually fully resolve the problem (since I don't know what's causing it) being clearer about where the config file is meant to be can only be a Good Thing.

Addresses #22104

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/UrlRewriter.java [hunk 1]
```diff
@@ -94,6 +94,21 @@ public static UrlRewriter getDownloaderUrlRewriter(String configPath, Reporter r
       return new UrlRewriter(log, "", new StringReader(""));
     }
 
+    // If the `configPath` is absolute, use that. Otherwise, prepend the `workspaceRoot`.
+    // There have been reports (eg. https://github.com/bazelbuild/bazel/issues/22104) that
+    // there are occasional errors when `configFile` can't be found, and when this happens
+    // investigation suggests that the current working directory isn't the workspace root.
+    Path actualConfigPath;
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/UrlRewriter.java:101]: this segment could simply be replaced with `workspaceRoot.getRelative(configPath)` -- it deals with an absolute `configPath` automatically.

Comment (shs96c) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/UrlRewriter.java]: I did not know that. Thank you! Fixed

**Review by Wyverald — APPROVED** [review]

**Comment by keith** [comment]
@bazel-io flag

**Comment by Wyverald** [comment]
@bazel-io fork 7.2.1

**Comment by Wyverald** [comment]
@bazel-io fork 7.3.0

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.2.1 RC2](https://github.com/bazelbuild/bazel/releases/tag/7.2.1rc2). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.2.1rc2`. Thanks!

**Pull request closed without merging**